projects/igniteui-angular/src/lib/directives/grid-interaction/grid-interaction.directive.ts

```diff
@@ -0,0 +1,211 @@
+                        const rowIndex = parseInt(target.getAttribute('data-rowindex'), 10);
+                        const visibleIndex = parseInt(target.getAttribute('data-visibleindex'), 10);
+
+                        grid = this.extractGrid(rowIndex, rowNode);
+                        const cell = grid.getCellByColumnVisibleIndex(rowIndex, visibleIndex);
+
+                        if (grid) {
+                            const row = grid.getRowByIndex(rowIndex);
+                            this.emitEvents(row, rowNode, evt, interaction, isCellInteraction ? cell : null);
+                        }
+                    }
+                });
+                (isCellInteraction ? this.cellStartListenerFn : this.rowStartListenerFn).push(startFn);
+            });
+        }
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * Emit the start and end events
+     */
+    private emitEvents(row, rowNode, event, interaction, cell?) {
+        if (row && row.nativeElement === rowNode) {
+            const interactionElement = cell ? cell : row;
+            if (cell) {
+                this.onCellInteractionStart.emit({ cell: interactionElement, originalEvent: event });
+            } else {
+                this.onRowInteractionStart.emit({ row: interactionElement, originalEvent: event });
+            }
+            interaction.end.forEach(endEvent => {
+                const endFn = this.renderer.listen(interactionElement.nativeElement, endEvent, (evt) => {
+                    if (cell) {
+                        this.onCellInteractionEnd.emit({ cell: interactionElement, originalEvent: evt });
+                    } else {
+                        this.onRowInteractionEnd.emit({ row: interactionElement, originalEvent: evt });
+                    }
+                });
+                (cell ? this.cellEndListenerFn : this.rowEndListenerFn).push(endFn);
+            });
+        }
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * Check if the tag name is grid cell
+     */
+    private isGridCell(tag: string): boolean {
+        return tag.toLowerCase() === 'igx-grid-cell' ||
+            tag.toLowerCase() === 'igx-tree-grid-cell' ||
+            tag.toLowerCase() === 'igx-hierarchical-grid-cell';
+    }
+
+    /**
+     * @hidden
+     * @internal
+     * Get the grid which contains the grid interaction directive
+     */
+    private extractGrid(rowIndex: number, rowNode: any): IgxGridBaseDirective {
+        let grid: IgxGridBaseDirective;
+        if (this.rowIsland) {
+            // the directive is applied to igx-row-island and should interact with grid representing that island
+            const childGrids = this.rowIsland.rowIslandAPI.getChildGrids().filter(child => {
+                const row = child.getRowByIndex(rowIndex);
+                return row && row.nativeElement === rowNode;
+            });
+            grid = childGrids.length > 0 ? childGrids[0] : null;
```

Comment:
In what case would this be `null`?
AFAIS this is called when handling the start interaction events, which raises the question can we fire an event on a grid that is not there?

Reply:
It will be null, when interacting with different row island from the one with the directive. We have to make sure the event is emitted to the correct row/cell in the correct grid.